balancer-js/src/modules/relayer/relayer.module.ts

```diff
@@ -147,8 +148,70 @@ export class Relayer {
     ]);
   }
 
-  static encodeUnwrapERC4626(params: EncodeUnwrapERC4626Input): string {
-    return relayerLibrary.encodeFunctionData('unwrapERC4626', [
+  static encodeUnwrapWstETH(params: EncodeUnwrapWstETHInput): string {
+    return relayerLibrary.encodeFunctionData('unwrapWstETH', [
+      params.sender,
+      params.recipient,
+      params.amount,
+      params.outputReference,
+    ]);
+  }
+
+  static encodeUnwrap(
+    params: EncodeUnwrapInput,
+    linearPoolType: string
+  ): string {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    let unwrapType: any;
+
+    /**
+     * Other unwrap types available on BatchRelayerLibrary that does not seem to
+     * have a respective Linear pool type in the SDK:
+     * - unwrapCompoundV2
+     * - unwrapShareToken
+     * - unwrapUnbuttonToken
+     * - unwrapWstETH
+     */
+
+    switch (linearPoolType) {
+      case 'AaveLinear':
+        return this.encodeUnwrapAaveStaticToken({
+          staticToken: params.wrappedToken,
+          sender: params.sender,
+          recipient: params.recipient,
+          amount: params.amount,
+          toUnderlying: true,
+          outputReference: params.outputReference,
+        });
+      case 'ERC4626Linear':
+        unwrapType = 'unwrapERC4626';
+        break;
+      case 'EulerLinear':
+        unwrapType = 'unwrapEuler';
+        break;
+      case 'GearboxLinear':
+        unwrapType = 'unwrapGearbox';
+        break;
+      case 'ReaperLinear':
+        unwrapType = 'unwrapReaperVaultToken';
+        break;
+      case 'TetuLinear':
+        unwrapType = 'unwrapTetu';
+        break;
+      case 'YearnLinear':
+        unwrapType = 'unwrapYearn';
+        break;
+      case 'Linear':
+      case 'BeefyLinear':
```

Comment:
'BeefyLinear':
'MidasLinear':
'SiloLinear':
Can we check with Integrations (or similar) if these pool types can use ERC4626 or another pool type unwrap?